File: projekt3/Wolf.py
from Animal import *

class Wolf(Animal):
    __WOLF_STRENGTH__ = 9
    __WOLF_INITIATIVE__ = 5
    __COLOR__ = "darkgray"
    
    def __init__(self, row, col, world):
        super().__init__(row, col, self.__WOLF_STRENGTH__, self.__WOLF_INITIATIVE__, world)
    
    def getName(self):
      return "Wolf"
    
    def getColor(self):
        return self.__COLOR__
    

    def reproduce(self, row, col):
        self._world_.setOrganism(row, col, Wolf())

File: projekt2/src/proj/java/model/animal/Wolf.java
package proj.java.model.animal;

import java.awt.Color;

import proj.java.model.Animal;
import proj.java.model.World;

public class Wolf extends Animal{

    private static final int WOLF_STRENGTH = 9;
    private static final int WOLF_INITIATIVE = 5;

    @Override
    public Color getColor() {
        return Color.DARK_GRAY;
    }

    @Override
    public int getInitiative() {
        return WOLF_INITIATIVE;
    }

    @Override
    public int getStrength() {
        return WOLF_STRENGTH + getExtraStrength();
    }

    @Override
    protected void reproduce(World world, int row, int col) {
        world.setOrganism(row, col, new Wolf());
    }
}
